### DISCUSSION TITLE
Get the interpretation of an uninterpreted relation from a model

### DISCUSSION THREAD

**Discussion by nicola-gigante** [cvc5/cvc5#9261] [Q&A] [answered]
Is there a way in the C++ API to obtain the interpretation of an uninterpreted relation from the model of a satisfiable problem?

In my use case, I have some finite sorts (non-recursive datatypes), some n-ary relations over those sorts, and some quantified assertions (`exists x` and `forall x exists y ...`, typically). When the problem is satisfiable, I can use the `Solver::getValue()` method to get the value of a term. But I do not need to know the value of any variable, what I need is to know which tuples are contained in the relations. Since the sorts are finite and enumerated, I can enumerate all the possible tuples and test with `getValue()` if `P(a,b,c)` holds for every `a`, `b`, and `c` of the right sorts, but that's exponentially long!

So is there a way to directly get which elements are contained in the interpretation of a relation?

**Comment by ajreynol** [comment, marked as answer]
You should be able to call `getValue` on the relation itself.  Would this suffice?

Note this discussion is related: https://github.com/cvc5/cvc5/discussions/9086

**Reply by nicola-gigante** [reply]
Hi! Thanks for the answer. The linked discussion is related indeed, mine is almost a duplicate! I didn't know it was possible to call `getValue()` on the relation itself. Which term do I get as a result and how do I inspect it? The other discussion talks about the "ite structure" of the relation. How does it work?

**Comment by ajreynol** [comment]
The resulting term should have kind `LAMBDA` (first child is the variable list, second child is its body).  To know what tuples are in the relation, you can evaluate the body while setting the arguments to the tuple to test.
